include/wx/osx/core/colour.h

```diff
@@ -29,60 +29,78 @@ class WXDLLIMPEXP_CORE wxColour: public wxColourBase
     // default copy ctor and dtor are ok
 
     // accessors
-    virtual bool IsOk() const { return m_cgColour != NULL; }
+    virtual bool IsOk() const wxOVERRIDE;
 
-    virtual WXDLLIMPEXP_INLINE_CORE ChannelType Red() const { return m_red; }
-    virtual WXDLLIMPEXP_INLINE_CORE ChannelType Green() const { return m_green; }
-    virtual WXDLLIMPEXP_INLINE_CORE ChannelType Blue() const { return m_blue; }
-    virtual WXDLLIMPEXP_INLINE_CORE ChannelType Alpha() const { return m_alpha; }
+    virtual ChannelType Red() const wxOVERRIDE;
+    virtual ChannelType Green() const wxOVERRIDE;
+    virtual ChannelType Blue() const wxOVERRIDE;
+    virtual ChannelType Alpha() const wxOVERRIDE;
+
+    wxColour& operator=(const wxColour& col);
 
     // comparison
     bool operator == (const wxColour& colour) const;
-
     bool operator != (const wxColour& colour) const { return !(*this == colour); }
 
-    CGColorRef GetPixel() const { return m_cgColour; }
+    // CoreGraphics CGColor
+    // --------------------
 
-    CGColorRef GetCGColor() const { return m_cgColour; }
-    CGColorRef CreateCGColor() const { return wxCFRetain( (CGColorRef)m_cgColour ); }
-
-#if wxOSX_USE_COCOA_OR_CARBON
-    void GetRGBColor( RGBColor *col ) const;
-#endif
-
-    // Mac-specific ctor and assignment operator from the native colour
-    // assumes ownership of CGColorRef
+    // This ctor does take ownership of the color.
     wxColour( CGColorRef col );
```

Comment:
I think this ctor (as well as the one taking `RGBColor` below) should be `explicit` to avoid bad surprises in both the library and (worse) user code,

Reply:
actually I wanted those to remove the need for assignment operators, I also don't see any dangerous implicit conversions to any of those, or am I overlooking the obvious ?

Reply:
Nothing really obvious, but I just thought that presence of implicit platform-specific conversions might be dangerous/unexpected.